grpc-example/grpc-example-scala/hello-api/src/main/scala/com/example/hello/api/HelloService.scala

```diff
@@ -18,7 +18,7 @@ trait HelloService extends Service {
 
   override final def descriptor = {
     import Service._
-    named("hello-srvc")
+    named("helloworld.GreeterService")
```

Comment:
This Shouldn't be necessary. IIRC, it is possible to force what name to use on the lookup for a gRPC connection.

Instead, I'd extract `"hello-svc"` to a constant and use that when [creating the `GrpcClientSettings`](https://github.com/lagom/lagom-samples/blob/1.6.x/grpc-example/grpc-example-scala/hello-proxy-impl/src/main/scala/com/example/helloproxy/impl/HelloProxyLoader.scala#L41). 

If the Lagom service has never been deployed both solution are OK. If the Lagom service is already in production, changin the name will break the exiting lagom clients that will be forced to upgrade and redeploy.

Reply:
Can you explain how the k8s service discovery is intended to work in this scenario?

Reply:
Sure.

The `named("hello-srvc")` is used in two places:
 - lagom dev.mode (not relevant right now)
 - lagom client --> the lagom client will extract the name (`"hello-srvc"`) and use that for a service location query.

If the Lagom service is already in production, it's likely there are also some services on the wild that expect the name to locate the service to be `"foo"` so be can't change that.

Then, when we add a new router to include gRPC support, the code generation and the default `GrpcClientSettings` default to using `"helloworld.GreeterService"` for lookups. 

And brings the problem: when deploying on k8s under what name should I register the service. What's important is to avoid the anti-pattern to register the same pod under two names. 

So, instead of changing the name on the Lagom service descriptor, if we change the name on the `GrpcClientSettings` what happens is this:

 - the akka-grpc/play-grpc code generators set the value to `"helloworld.GreeterService"`
 - the service lookup prioritizes the value on `GrpcClientSettings` and issues a `DnsSrv("hello-srvc")` query instead.

PS. the other concern is the `portName` field on the `DnsSrv(...)` query (see my comment on naming the [ports in the YAML](https://github.com/lagom/lagom-samples/pull/166#pullrequestreview-420499452) and the creation of the `GrpcClientSettings`.

Reply:
> Can you explain how the k8s service discovery is intended to work in this scenario?

Looking at the last leg of the equation:

 - in the `Deployment` yaml the name of the service will have to be `hello-srvc`

That value is what k8s will use to identify all pods running this binary.

Reply:
PS: binaries with multiple names are a [PITA](https://github.com/lagom/lagom/issues/580). Pick one stable name.